hw3/02-charts.js

```diff
@@ -1,52 +1,119 @@
+  "rgba(210, 199, 199, 0.8)",
+  "rgba(78, 52, 199, 0.8)",
 ];
 
 const borderColors = [
-  'rgba(54, 162, 235, 1)',
-  'rgba(255, 206, 86, 1)',
-  'rgba(255, 99, 132, 1)',
-  'rgba(75, 192, 192, 1)',
-  'rgba(153, 102, 255, 1)',
-  'rgba(255, 159, 64, 1)',
-  'rgba(159, 159, 159, 1)',
-  'rgba(83, 102, 255, 1)',
-  'rgba(40, 159, 64, 1)',
-  'rgba(210, 199, 199, 1)',
-  'rgba(78, 52, 199, 1)',
+  "rgba(54, 162, 235, 1)",
+  "rgba(255, 206, 86, 1)",
+  "rgba(255, 99, 132, 1)",
+  "rgba(75, 192, 192, 1)",
+  "rgba(153, 102, 255, 1)",
+  "rgba(255, 159, 64, 1)",
+  "rgba(159, 159, 159, 1)",
+  "rgba(83, 102, 255, 1)",
+  "rgba(40, 159, 64, 1)",
+  "rgba(210, 199, 199, 1)",
+  "rgba(78, 52, 199, 1)",
 ];
 
 // url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
-
-const renderChart = () => {
-  const donutChart = document.querySelector('.donut-chart');
+const url = "https://thronesapi.com/api/v2/Characters";
+const donutChart = document.querySelector(".donut-chart");
+let housesObject = {};
 
+function cleanfamilyName(family) {
+  family = family.replace(/House/, "").trim();
+  if (family.includes("Lanister")) family = "Lannister";
+  else if (family.includes("None")) family = "Unknown";
+  else if (family.includes("Unkown")) family = "Unknown";
+  else if (family.includes("Targaryan")) family = "Targaryen";
+  else if (family.includes("Targaryn")) family = "Targaryen";
+  else if (family === "") family = "Unknown";
+  return family;
+}
+function groupHouses(housesObject) {
+  let newHouseObject = {};
+  for (const key of Object.keys(housesObject)) {
+    if (housesObject[key] === 1) {
+      if ("Others" in newHouseObject) {
+        newHouseObject["Others"] = newHouseObject["Others"] + 1;
+      } else {
+        newHouseObject["Others"] = 1;
+      }
+    } else {
+      newHouseObject[key] = housesObject[key];
+    }
+  }
+  return newHouseObject;
+}
+const renderChart = (houseLabels, houseValues) => {
   new Chart(donutChart, {
-    type: 'doughnut',
+    type: "doughnut",
     data: {
-      labels: ['label', 'label', 'label', 'label'],
+      labels: houseLabels,
       datasets: [
         {
-          label: 'My First Dataset',
-          data: [1, 12, 33, 5],
+          label: "House of Dragons",
+          data: houseValues,
           backgroundColor: backgroundColors,
           borderColor: borderColors,
           borderWidth: 1,
         },
       ],
     },
+    options: {
+      title: {
+        display: true,
+        text: "Exercise 02 - Charts",
+        fontSize: 40,
+      },
+      plugins: {
+        labels: {
+          fontSize: 10,
+        },
+      },
+      legend: {
+        position: "bottom",
+      },
+    },
   });
 };
+const calculateHouseMembers = async function () {
+  try {
+    const response = await fetch(url);
+    const jsonData = await response.json();
+    jsonData.forEach((element) => {
+      let { family } = element;
+      family = cleanfamilyName(family);
```

Comment:
```suggestion
      const family = cleanfamilyName(element.family);
```

so we have less mutable (`let`)